### PR TITLE
Feature/61004 create endpoint to retrieve process link for start events

### PR DESCRIPTION
**Pull request by luis-soares-ritense**

### CODE REVIEW

File: process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/OpenProcessLinkResult.kt [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 package com.ritense.processlink.web.rest.dto
 
-data class OpenTaskResult<T> (
+data class OpenProcessLinkResult<T> (
```
Comment (theo-ritense) [inline process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/OpenProcessLinkResult.kt:19]: Not quite agreeing with this rename.
I think `ProcessLinkTaskResult` fits better.

Comment (luis-soares-ritense) [reply process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/OpenProcessLinkResult.kt:19]: well it's not only about opening a task anymore. it's to retrieve a form or form-flow as well

Comment (theo-ritense) [reply process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/OpenProcessLinkResult.kt:19]: `StartProcessLinkResult` then?

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/FormFlowProcessLinkActivityHandler.kt [hunk 2]
```diff
@@ -47,7 +47,18 @@ class FormFlowProcessLinkTaskProvider(
...
             else -> throw IllegalStateException("Multiple form flow instances linked to task: ${task.id}")
         }
-        return OpenTaskResult(FORM_FLOW_TASK_TYPE_KEY, FormFlowTaskOpenResultProperties(instance.id.id))
+        return OpenProcessLinkResult(FORM_FLOW_TASK_TYPE_KEY, FormFlowTaskOpenResultProperties(instance.id.id))
+    }
+
+    override fun getStartEventObject(processDefinitionId: String, processLink: ProcessLink): OpenProcessLinkResult<FormFlowTaskOpenResultProperties> {
+        processLink as FormFlowProcessLink
+        val formFlowDefinition = formFlowService.findDefinition(processLink.formFlowDefinitionId)!!
+        val additionalProperties = mapOf<String,Any>(Pair("processDefinitionId",processDefinitionId))
```
Comment (Klaas-Ritense) [inline form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/FormFlowProcessLinkActivityHandler.kt:56]:
```suggestion
        val additionalProperties = mapOf<String,Any>("processDefinitionId" to processDefinitionId)
```

File: process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/OpenProcessLinkResult.kt (hunk 1 above)
Comment (marijnritense) [reply process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/OpenProcessLinkResult.kt:19]: It's used for 2 things:
1. Use a process link to open a task
2. Use a process link to open a start event
Both of those are activities, so I'd say `ProcessLinkActivityResult` makes most sense if we want to stick to the camunda naming.

Comment (luis-soares-ritense) [reply process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/OpenProcessLinkResult.kt]: agreed. Done!

**Pull request merged**